exchanges/poloniex/poloniex_test.go

```diff
@@ -141,901 +71,2308 @@ func TestGetFee(t *testing.T) {
-	sharedtestvalues.SkipTestIfCredentialsUnset(t, e)
-
-	_, err := e.GetWithdrawalsHistory(t.Context(), currency.BTC, asset.Spot)
-	if err != nil {
-		t.Error(err)
+	orderStringMap := map[order.Type]struct {
+		String string
+		Error  error
+	}{
+		order.Market:       {String: order.Market.String()},
+		order.Limit:        {String: order.Limit.String()},
+		order.LimitMaker:   {String: order.LimitMaker.String()},
+		order.StopLimit:    {String: "STOP_LIMIT"},
+		order.AnyType:      {},
+		order.UnknownType:  {},
+		order.TrailingStop: {Error: order.ErrUnsupportedOrderType},
+	}
+	for k, v := range orderStringMap {
+		result, err := OrderTypeString(k)
+		require.ErrorIs(t, err, v.Error)
+		assert.Equal(t, v.String, result)
 	}
 }
 
-func TestCancelBatchOrders(t *testing.T) {
+func TestStringToOrderType(t *testing.T) {
 	t.Parallel()
-	sharedtestvalues.SkipTestIfCredentialsUnset(t, e, canManipulateRealOrders)
-	_, err := e.CancelBatchOrders(t.Context(), []order.Cancel{
-		{
-			OrderID:   "1234",
-			AssetType: asset.Spot,
-			Pair:      currency.NewBTCUSD(),
-		},
-	})
-	if err != nil {
-		t.Error(err)
+	orderTypeStringToTypeMap := map[string]order.Type{
+		"":                    order.Limit,
+		"STOP":                order.Stop,
+		"STOP_LIMIT":          order.StopLimit,
+		"TRAILING_STOP":       order.TrailingStop,
+		"TRAILING_STOP_LIMIT": order.TrailingStopLimit,
+	}
+	for k, v := range orderTypeStringToTypeMap {
+		result := StringToOrderType(k)
+		assert.Equal(t, result, v)
 	}
 }
 
-func TestGetTimestamp(t *testing.T) {
+func TestOrderStateString(t *testing.T) {
 	t.Parallel()
-	st, err := e.GetTimestamp(t.Context())
-	require.NoError(t, err)
-
-	if st.IsZero() {
-		t.Error("expected a time")
+	orderStatusToStringMap := map[string]order.Status{
+		"NEW":                order.New,
+		"FAILED":             order.Closed,
+		"FILLED":             order.Filled,
+		"CANCELED":           order.Cancelled,
+		"abcd":               order.UnknownStatus,
+		"PARTIALLY_FILLED":   order.PartiallyFilled,
+		"PARTIALLY_CANCELED": order.PartiallyCancelled,
+	}
+	for k, v := range orderStatusToStringMap {
+		result := orderStateFromString(k)
+		assert.Equal(t, v, result)
 	}
 }
 
-func TestGetServerTime(t *testing.T) {
+func TestStringToOrderSide(t *testing.T) {
 	t.Parallel()
-	st, err := e.GetServerTime(t.Context(), asset.Spot)
-	require.NoError(t, err)
-
-	if st.IsZero() {
-		t.Error("expected a time")
+	stringToOrderSideMap := map[string]order.Side{
+		order.Sell.String():  order.Sell,
+		order.Buy.String():   order.Buy,
+		order.Short.String(): order.Short,
+		order.Long.String():  order.Long,
+		"":                   order.UnknownSide,
+	}
+	for k, v := range stringToOrderSideMap {
+		result := stringToOrderSide(k)
+		assert.Equal(t, v, result)
 	}
 }
 
-func TestFetchTradablePairs(t *testing.T) {
-	t.Parallel()
-	_, err := e.FetchTradablePairs(t.Context(), asset.Spot)
-	if err != nil {
-		t.Error(err)
+func generateContext() context.Context {
```

Comment:
```go
func generateContext() context.Context {
	if mockTests {
		return account.DeployCredentialsToContext(context.Background(), &account.Credentials{
			Key:    "abcde",
			Secret: "fghij",
		})
	}
	return context.Background()
}
```

Could even take in the testing context so that its cancellations propagate. And then change name.